src/SAML2/Assertion.php

```diff
@@ -1462,6 +1493,16 @@ private function addAttributeStatement(DOMElement $root)
                 $attribute->setAttribute('NameFormat', $this->nameFormat);
             }
 
+            if ($name === SAML2_Const::EPTI_URN_MACE || $name === SAML2_Const::EPTI_URN_OID) {
+                foreach ($values as $eptiValue) {
+                    $attributeValue = $document->createElementNS(SAML2_Const::NS_SAML, 'saml:AttributeValue');
+                    $attribute->appendChild($attributeValue);
+                    SAML2_Utils::addNameId($attributeValue, $eptiValue);
```

Comment:
It should now be resolved in all branches (master, 1.x & 2.x). Could you please verify that the fix doesn't break anything else?

The fix in 1.x and 2.x is a bit different than in master. Basically, because in master it is a much more complete fix, using the `SAML2\XML\saml\NameID` class instead of an array. However, this breaks backwards-compatibility too, so we need to evaluate it carefully.